src/roles/actors.rs

```diff
@@ -212,19 +212,29 @@ impl<T: Trait> Module<T> {
         role: Role,
         member_id: MemberId<T>,
         unbonding_period: T::BlockNumber,
+        stake: BalanceOf<T>,
     ) {
         // simple unstaking ...only applying unbonding period
-        let value = T::Currency::free_balance(&actor_account);
-        T::Currency::set_lock(
-            STAKING_ID,
+        Self::update_lock(
             &actor_account,
-            value,
+            stake,
             <system::Module<T>>::block_number() + unbonding_period,
-            WithdrawReasons::all(),
         );
 
         Self::remove_actor_from_service(actor_account, role, member_id);
     }
+
+    // Locks account and prevents transfers and reservation. Account still pay basic
+    // transaction fees and therefore send transactions that don't demand reserving balance
+    fn update_lock(account: &T::AccountId, stake: BalanceOf<T>, until: T::BlockNumber) {
+        T::Currency::set_lock(
+            STAKING_ID,
+            account,
+            stake,
+            until,
+            WithdrawReason::Transfer | WithdrawReason::Reserve,
```

Comment:
I don't know all the possible `WithdrawReason` values, so trust you here.

This is a little brittle with regards to changes, if e.g. more `WithdrawReason` values are added. You could start with `::all()` and exclude transaction fees, perhaps?

Reply:
e.g.
```rust
WithdrawReason::all() & ~WithdrawReason:TransactionFee // or whatever the reason is called
```

Reply:
good point, will fix